src/libstore/content-address.hh

```diff
@@ -39,12 +39,26 @@ enum struct FileIngestionMethod : uint8_t {
     /**
      * Flat-file hashing. Directly ingest the contents of a single file
      */
-    Flat = false,
+    Flat,
+
     /**
      * Recursive (or NAR) hashing. Serializes the file-system object in Nix
      * Archive format and ingest that
      */
-    Recursive = true
+    Recursive,
+
+    /**
+     * Git hashing. In particular files are hashed as git "blobs", and
+     * directories are hashed as git "trees".
+     *
+     * @note Git's data model is slightly different, in that a plain
+     * fail doesn't have an executable bit, directory entries do
+     * instead. We decide treat a bare file as non-executable by fiat,
```

Comment:
```suggestion
     * instead. We decide treat a bare file as non-executable by default,
```

Reply:
```suggestion
     * instead. We decide treat a bare file as non-executable.
```

Reply:
I misunderstood "by fiat" can be used as an expression like this, and confused it for a typo.